test/simulation/xudtest/node.go

```diff
@@ -253,6 +253,18 @@ func (hn *HarnessNode) ConnectRPC(useMacs bool) (*grpc.ClientConn, error) {
 	}
 
 	tlsCreds, err := credentials.NewClientTLSFromFile(hn.Cfg.TLSCertPath, "")
+	var tries = 3
+	for tries > 0 {
+		if err != nil {
```

Comment:
Shouldn't this be `if err == nil` ? We only want to retry on errors.

Or we could check and break up front without the need for if/else.

```go
if err == nil {
	break;
}
// retry logic below
```